compiler/astutil/astutil.go

```diff
@@ -102,15 +94,61 @@ func PruneOriginal(d *ast.FuncDecl) bool {
 // function in the original called `foo`, it will be accessible by the name
 // `_gopherjs_original_foo`.
 func KeepOriginal(d *ast.FuncDecl) bool {
-	if d.Doc == nil {
-		return false
-	}
-	for _, c := range d.Doc.List {
-		if strings.HasPrefix(c.Text, "//gopherjs:keep-original") {
-			return true
+	return hasDirective(d, `keep-original`)
+}
+
+// anyDocLine calls the given predicate on all associated documentation
+// lines and line-comment lines from the given node.
+// If the predicate returns true for any line then true is returned.
+func anyDocLine(node any, predicate func(line string) bool) bool {
```

Comment:
Optional: https://pkg.go.dev/go/ast#Inspect may be a more concise alternative to manual AST traversal.

Reply:
Awesome! I hadn't considered/noticed `Inspect`. I'll update to use that instead.

Reply:
Update. That looks better. I added a few more tests to make sure if we are looking for directives on declarations we don't accidentally dig down into the specifications and get a false positive.

Essentially, I wanted this to work:

```Go
//gopherjs:do-stuff
type (           // <-- has do-stuff
   Bill struct{} // <-- has no directive (inherits do-stuff in augmenter)
   Bob  struct{} // <-- has no directive (inherits do-stuff in augmenter)
)

type (            // <-- has no directive
   //gopherjs:do-stuff
   Sally struct{} // <-- has do-stuff  

   //gopherjs:do-other-thing
   Larry struct{} // <-- has do-other-thing
)

//gopherjs:do-stuff
type Foo struct{} // <-- the declaration has do-stuff, Foo does not but inherits it in augmenter
```